core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala

```diff
@@ -17,37 +17,47 @@
 
 package kafka.admin
 
-import java.io.{ByteArrayOutputStream, PrintStream}
-import java.nio.charset.StandardCharsets
-import scala.collection.Seq
 import kafka.integration.KafkaServerTestHarness
 import kafka.server.KafkaConfig
-import kafka.utils.TestUtils
+import kafka.utils.{TestInfoUtils, TestUtils}
 import org.apache.kafka.clients.NodeApiVersions
+import org.apache.kafka.common.message.ApiMessageType
 import org.apache.kafka.common.protocol.ApiKeys
 import org.junit.jupiter.api.Assertions.{assertEquals, assertFalse, assertNotNull, assertTrue}
-import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.Timeout
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
 
+import java.io.{ByteArrayOutputStream, PrintStream}
+import java.nio.charset.StandardCharsets
+import scala.collection.Seq
 import scala.jdk.CollectionConverters._
 
 class BrokerApiVersionsCommandTest extends KafkaServerTestHarness {
 
   def generateConfigs: Seq[KafkaConfig] =
-    TestUtils.createBrokerConfigs(1, zkConnect).map(props => {
-      // Configure control plane listener to make sure we have separate listeners from client,
-      // in order to avoid returning Envelope API version.
-      props.setProperty(KafkaConfig.ControlPlaneListenerNameProp, "CONTROLLER")
-      props.setProperty(KafkaConfig.ListenerSecurityProtocolMapProp, "CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT")
-      props.setProperty("listeners", "PLAINTEXT://localhost:0,CONTROLLER://localhost:0")
-      props.setProperty(KafkaConfig.AdvertisedListenersProp, "PLAINTEXT://localhost:0,CONTROLLER://localhost:0")
-      props.setProperty(KafkaConfig.UnstableApiVersionsEnableProp, "true")
-      props
-    }).map(KafkaConfig.fromProps)
+    if (isKRaftTest()) {
+      TestUtils.createBrokerConfigs(1, null).map(props => {
+        props.setProperty(KafkaConfig.UnstableApiVersionsEnableProp, "true")
```

Comment:
Please add a comment about this change, I guess this is related to KIP-848, and we can remove this after KIP-848 is all set.